File: android/src/main/java/com/reactnativecrispchatsdk/parsers/EmploymentParser.kt
package com.reactnativecrispchatsdk.parsers

import com.facebook.react.bridge.ReadableMap
import com.reactnativecrispchatsdk.getStringOrNull
import im.crisp.client.external.data.Employment

/**
 * Parser for Employment objects from JavaScript
 * Converts ReadableMap to Crisp SDK Employment class
 *
 * TypeScript interface:
 * interface Employment {
 *   title?: string;
 *   role?: string;
 * }
 */
object EmploymentParser {
    fun fromReadableMap(map: ReadableMap): Employment {
        val title = map.getStringOrNull("title")
        val role = map.getStringOrNull("role")
        return Employment(title, role)
    }
}

File: ios/Parsers/EmploymentParser.swift
import Foundation
import Crisp

@objc
public class EmploymentParser: NSObject {

    @objc
    public static func fromDictionary(_ dict: NSDictionary) -> Employment {
        let title = dict.getStringOrNil(forKey: "title")
        let role = dict.getStringOrNil(forKey: "role")
        return Employment(title: title, role: role)
    }
}
